src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -52,6 +52,26 @@ def _attach_related(_events):
             event.group = groups.get(event.group_id)
         eventstore.bind_nodes(_events, "data")
 
+    def _get_grouping_variants(event):
+        from sentry.grouping.api import get_grouping_variants_for_event, load_grouping_config
+        from sentry.stacktraces.processing import normalize_stacktraces_for_grouping
+
+        config = event.data.get("grouping_config")
+        config = load_grouping_config(config)
+        return get_grouping_variants_for_event(event, config)
+
+    def _get_hashes(event):
+        hashes = event.data.get("hashes")
+        if hashes is not None:
+            return hashes
+
+        return filter(
+            None, [x.get_hash() for x in _get_grouping_variants(event).values()]
```

Comment:
Since we're never gonna use anything except the first not-None value, we may wanna use `next` and a clever generator here too (only if the grouping variants can be long):

```py
return next(hash for hash in (x.get_hash() for x in _get_grouping_variants(event).itervalues()))
```

This should give us the best memory efficiency.